### DISCUSSION TITLE
Suggestion: remove custom properties from base styles

### DISCUSSION THREAD

**Discussion by felquis** [tailwindlabs/tailwindcss#10290] [Ideas]
Hi everyone, I think this suggest could make tailwind even more configurable.

The `@tailwind base;` line today comes with a CSS reset AND a few custom properties.

Problems with that:
- if you need to use transform or filter css classes, you must include the base styles because these classes depends on the custom properties.
- a project that doesn't want to add the base styles will eventually get to the above error.
- A ongoing product that started without tailwindcss may want to introduce tailwind but removing the reset makes it easier to use tailwind.
- but when removing the base styles, it also breaks the transform/filter classes.

Solution:
- Keep `@tailwind base;` as is, but:
- Introduce `@tailwind reset`
- Introduce `@tailwind custom-properties`

This way, a developer like myself can keep tailwind custom properties and get rid of the reset while ignoring the base styles completely.

Advantages of suggested solution:
- backward compatibility, people need to change zero code that they already use
- on going products that want to use tailwind can easily opt-out from tailwind default reset.

Conclusion, that's it, let me know how do you feel about this suggestion, have you ever ran into this problem?

Thank you so very much.

**Comment by wongjn** [comment]
You can disable the reset without removing the custom property defaults by [disabling the preflight plugin](https://tailwindcss.com/docs/preflight#disabling-preflight).

**Reply by aenial51** [reply]
Ok I will try as per given your suggestion.
[Nexus Iceland
](https://www.nexus-iceland.com/)
